### ISSUE TITLE
I am having issues to build the latest version fo Keywhiz...

### ISSUE THREAD

**Issue by FlavioImbertDomingos** [square/keywhiz#598]
Hi There,

Sorry to buggy you guys, but I am having issues to build the latest version. I am not a Java developer, so I may be missing something. My goal is to build this binary 0.10.2 and deploy/upgrade in our cluster running 0.8.0.

So, this is the issue when I build and maven run the tests:

[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Keywhiz (Parent) 0.10.2-SNAPSHOT:
[INFO] 
[INFO] Keywhiz (Parent) ................................... SUCCESS [  2.032 s]
[INFO] Keywhiz Testing .................................... SUCCESS [  2.732 s]
[INFO] Keywhiz API ........................................ SUCCESS [ 11.016 s]
[INFO] Keywhiz Client ..................................... SUCCESS [  0.445 s]
[INFO] Keywhiz CLI ........................................ SUCCESS [ 11.427 s]
[INFO] Keywhiz HKDF ....................................... SUCCESS [  1.902 s]
[INFO] Keywhiz Model ...................................... FAILURE [  0.798 s]
[INFO] Keywhiz Log ........................................ SKIPPED
[INFO] Keywhiz Server ..................................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  30.577 s
[INFO] Finished at: 2020-02-17T21:17:38Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.flywaydb:flyway-maven-plugin:6.1.3:migrate (default) on project keywhiz-model: org.flywaydb.core.internal.license.FlywayEditionUpgradeRequiredException: Flyway Enterprise Edition or MariaDB upgrade required: MariaDB 10.0 is no longer supported by Flyway Community Edition, but still supported by Flyway Enterprise Edition. -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :keywhiz-model
[root@keywhiz-233048331-1-258246470 keywhiz]# 

And if I try to run the compiled version keywhiz-hkdf-0.10.2-SNAPSHOT.jar, I am getting the following:

$ java -jar  keywhiz-hkdf-0.10.2-SNAPSHOT.jar  
no main manifest attribute, in keywhiz-hkdf-0.10.2-SNAPSHOT.jar

Do you know what I have to do to fix those issues above?

Thank you so much,

Flavio.

**Comment by mbyczkowski** [comment]
Hi Flavio,

We have bumped the migration library, Flyway, from 5.x to 6.x and it raised the minimal version requirements for some databases. You ran into this as well and realized we needed MySQL 5.7 or higher (and we have updated the README accordingly).

Looking at your logs, it says `org.flywaydb.core.internal.license.FlywayEditionUpgradeRequiredException: Flyway Enterprise Edition or MariaDB upgrade required: MariaDB 10.0 is no longer supported by Flyway Community`. 

I'm not super familiar with MariaDB, but version 10.0 was EOLd in March 2019, so maybe try at least 10.1?

**Comment by alokmenghrajani** [comment]
You'll have to either upgrade your database (the recommended path forward) or downgrade Flyway (a valid temporary approach).

**Comment by FlavioImbertDomingos** [comment]
Thanks Alok!

Yeah, I got it right now.

I had to install the latest Mariadb 10.4.12 and I was able to build and pass the automate tests now. 

[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Keywhiz (Parent) 0.10.2-SNAPSHOT:
[INFO] 
[INFO] Keywhiz (Parent) ................................... SUCCESS [  2.639 s]
[INFO] Keywhiz Testing .................................... SUCCESS [  3.415 s]
[INFO] Keywhiz API ........................................ SUCCESS [ 13.529 s]
[INFO] Keywhiz Client ..................................... SUCCESS [  0.861 s]
[INFO] Keywhiz CLI ........................................ SUCCESS [ 13.484 s]
[INFO] Keywhiz HKDF ....................................... SUCCESS [  2.593 s]
[INFO] Keywhiz Model ...................................... SUCCESS [ 13.106 s]
[INFO] Keywhiz Log ........................................ SUCCESS [  0.170 s]
[INFO] Keywhiz Server ..................................... SUCCESS [02:13 min]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  03:03 min
[INFO] Finished at: 2020-02-18T19:13:38Z
[INFO] ------------------------------------------------------------------------
[root@keywhiz-build-vm-54822395-2-879539820 keywhiz]# 

Now, i will test my upgrade from 0.8 to 0.10.2


Thank you!

Flavio.
